tests/unit/test_extras.hxx

```diff
@@ -195,16 +198,32 @@ public:
     output_info.disable();
     mesh->createDefaultRegions();
     output_info.enable();
+
+    // Delete any existing mesh_staggered
+    if (mesh_staggered != nullptr) {
+      delete mesh_staggered;
+      mesh_staggered = nullptr;
+    }
+    mesh_staggered = new FakeMesh(nx, ny, nz);
+    mesh_staggered->StaggerGrids = true;
+    output_info.disable();
+    dynamic_cast<FakeMesh*>(mesh_staggered)->setCoordinates(nullptr, CELL_XLOW);
```

Comment:
I think this can be a `static_cast` as we know all the types are fixed in advance?